### DISCUSSION TITLE
send X-Userinfo to upstream

### DISCUSSION THREAD

**Discussion by wkharrat** [apache/apisix#7102] [Q&A]
when the plugin openidc send X-Userinfo to upstream, is it in clear format or encoded?
can associate function to retrieve some information from the json of the header before sendin it to the backend, for exemple if i want to retrieve preffered_username

**Comment by tokers** [comment]
> can associate function to retrieve some information from the json of the header before sendin it to the backend, for exemple if i want to retrieve preffered_username

What does this mean? Who is I? You mean fetch the `preffered_username` in the internal of APISIX?

**Reply by wkharrat** [reply]
may be apisix cans send X-Userinfo to the upstream. Can i read the content of this header before sending it, (using an existing plugin), extract user name info, and send it(to the upstream) in new header that i create in the funtion code.

**Reply by tzssangglass** [reply]
But I don't remember a plugin that does this directly.

You can write your own plugins to implement this, with the relevant code scattered in various places.
